docs/deployment/automated-local.rst

```diff
@@ -0,0 +1,227 @@
+
+	# Archive name schema.
+	# If you intend to use prune and have more than one backup set, perhaps
+	# swap hostname and date to be compliant with prune's --prefix option.
+	DATE=$(date --iso-8601)-$(hostname)
+
+	# This is the file that will later contain UUIDs of registered backup drives
+	DISKS=/etc/backups/backup.disks
+
+	# Find whether the connected block device is a backup drive
+	for uuid in $(lsblk -nlpo uuid)
+	do
+	        if grep -qF $uuid $DISKS; then
+	                break
+	        fi
+	        uuid=
+	done
+
+	if [ ! $uuid ]; then
+	        echo "No backup disk found, exiting"
+	        exit 0
+	fi
+
+	echo "Disk $uuid is a backup disk"
+	disk_path=/dev/disk/by-uuid/$uuid
+	# Mount file system if not already done. This assumes that if something is already
+	# mounted at $MOUNTPOINT, it is the backup drive. It won't find the drive if
+	# it was mounted somewhere else.
+	(mount | grep $MOUNTPOINT) || mount $disk_path $MOUNTPOINT
+	drive=$(lsblk -snlpo name $disk_path | head -n1)
+	echo "Drive path: $drive"
+
+	#
+	# Create backups
+	#
+
+	# Options for borg create
+	BORG_OPTS="-svx --compression lz4 --checkpoint-interval 86400"
+
+	# Set BORG_PASSPHRASE or BORG_PASSCOMMAND somewhere around here, using export,
+	# if encryption is used.
+
+	# No one can answer if Borg asks these questions, it is better to just fail quickly
+	# instead of hanging.
+	export BORG_RELOCATED_REPO_ACCESS_IS_OK=no
+	export BORG_UNKNOWN_UNENCRYPTED_REPO_ACCESS_IS_OK=no
+
+	# Log Borg version
+	borg -V
+
+	echo "Starting backup for $DATE"
+
+	borg create $BORG_OPTS \
+	  --exclude /root/.cache \
+	  --exclude /var/cache \
+	  --exclude /var/lib/docker/devicemapper \
+	  $TARGET::$DATE-$$-system \
+	  / /boot
+
+	borg create $BORG_OPTS \
+	  --exclude 'sh:/home/*/.cache' \
+	  $TARGET::$DATE-$$-system \
+	  /home/
+
+	echo "Completed backup for $DATE"
+
+	# Just to be completely paranoid
+	sync
+
+	if [ -f /etc/backups/autoeject ]; then
+	        umount $MOUNTPOINT
+	        hdparm -Y $drive
+	fi
+
+	if [ -f /etc/backups/backup-suspend ]; then
+	        systemctl suspend
+	fi
+
+Create the ``/etc/backups/autoeject`` file to have the script automatically eject the drive
+after creating the backup. Rename the file to something else (e.g. ``/etc/backup/autoeject-no``)
+when you want to do something with the drive after creating backups (e.g running check).
+
+Create the ``/etc/backups/backup-suspend`` file if the machine should suspend after completing
+the backup. Don't forget to physically disconnect the device before resuming,
+otherwise you'll enter a cycle. You can also add an option to power down instead.
+
+Create an empty ``/etc/backups/backup.disks`` file, you'll register your backup drives
+there.
+
+The last part is to actually enable the udev rules and services:
+
+.. code-block:: bash
+
+	ln -s /etc/backups/40-backup.rules /etc/udev/rules.d/40-backup.rules
+	ln -s /etc/backups/automatic-backup.service /etc/systemd/system/automatic-backup.service
```

Comment:
I don't think linking from these directories to random other places is a good idea. It does keep things in one place though.